### PR TITLE
Feature/lef 228 migrate sharepoint api module over to

### PR DESCRIPTION
**Pull request by roboli**
Some observations:

* I decided to omit testing Auth flow since this better be tested at the parent class, otherwise every module will have to test same functionality again and again, unless there are some specifics to current api module but I don't see that with this one.

* Same with methods, tests are only for current class/instance, other methods should be tested with parent class tests.

* I decided to use `nock` instead of `msw` since `nock` is good enough for mocking and `msw` basically replaces a backend server. So `nock` lighter IMHO.

### CODE REVIEW

File: api-module-library/sharepoint/api.js [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                `/drives/${driveId}/items/${fileId}?$expand=listItem`,
+            search: ({ driveId, query }) =>
+                `/drives/${driveId}/root/search(q='${query}')?top=20&$select=id,image,name,file,parentReference,size,lastModifiedDateTime,@microsoft.graph.downloadUrl&$filter=`,
+        };
+
+        this.authorizationUri = `https://login.microsoftonline.com/${this.tenant_id}/oauth2/v2.0/authorize`;
+        this.tokenUri = `https://login.microsoftonline.com/${this.tenant_id}/oauth2/v2.0/token`;
+    }
+
+    getAuthUri() {
```
Comment (roboli) [inline api-module-library/sharepoint/api.js:42]: I removed `async` from here

File: api-module-library/sharepoint/manager.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+                name,
+                externalId,
+            };
+            this.entity = await Entity.create(createObj);
+        } else if (search.length === 1) {
+            this.entity = search[0];
+        } else {
+            const message = 'Multiple entities found with the same external ID: ' + externalId;
+            debug(message);
+            throw new Error(message);
```
Comment (roboli) [inline api-module-library/sharepoint/manager.js:116]: An error is being thrown now.

And replaced the `portalId` reference which came from nowhere, with `externalId`

**Review by roboli — COMMENTED** [review]
Left some comments @seanspeaks

File: api-module-library/sharepoint/LICENSE.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+MIT License
+
+Copyright (c) 2022 Left Hook Inc.
```
Comment (seanspeaks) [inline api-module-library/sharepoint/LICENSE.md:3]: Can you modify to 2023?

File: api-module-library/sharepoint/README.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
+# salesforce
+
+This is the API Module for salesforce that allows the [Frigg](https://friggframework.org) code to talk to the salesforce API.
```
Comment (seanspeaks) [inline api-module-library/sharepoint/README.md:3]: This was a copy/paste oopsie

Should say Microsoft SharePoint

File: api-module-library/sharepoint/README.md [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+Read more on the [Frigg documentation site](https://docs.friggframework.org/api-modules/list/salesforce
```
Comment (seanspeaks) [inline api-module-library/sharepoint/README.md:5]: You can set the link to microsoft-sharepoint (even though the page doesn't yet exist)

File: api-module-library/sharepoint/defaultConfig.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+    "name": "microsoft-sharepoint",
+    "label": "Microsoft SharePoint",
+    "productUrl": "https://microsoft.com/sharepoint",
+    "apiDocs": "Good Luck",
+    "logoUrl": "https://friggframework.org/assets/img/microsoft-sharepoint.jpeg",
+    "categories": ["Sharing"],
+    "description": "If you know, you know"
```
Comment (seanspeaks) [inline api-module-library/sharepoint/defaultConfig.json:8]: Can you grab Microsoft's description of SharePoint from somewhere? Maybe the H1 of the product site? One or two sentences is fine

File: api-module-library/sharepoint/jest-env [hunk 7]
```diff
@@ -0,0 +1,4 @@
+SHAREPOINT_SCOPE=sharepoint_scope_test
```
Comment (seanspeaks) [inline api-module-library/sharepoint/jest-env:1]: What's this file?

I think I want to move towards .env for actual test runs (loaded with dotenv) and .env.example for collaboration purposes

**Review by seanspeaks — CHANGES_REQUESTED** [review]
Some small changes, mostly documentation oriented. One functional in the processAuthorizationCallback method on the manager. 

Mostly looks great and really appreciate the thoroughness of the unit tests. 👌

File: api-module-library/sharepoint/manager.test.js [hunk 8]
```diff
@@ -0,0 +1,675 @@
...
+            });
+
+            const requirements = manager.getAuthorizationRequirements();
+            expect(requirements).toBeDefined();
+            expect(requirements.type).toEqual('oauth2');
+            expect(requirements.url).toEqual(`${manager.api.authorizationUri}?${queryParams}`);
+        });
+    });
+
+    describe('#processAuthorizationCallback', () => {
```
[the remark this answers is not in the document]
Comment (roboli) [reply api-module-library/sharepoint/manager.test.js:165]: Sorry, missed this @seanspeaks. Yeah, means a method test. I can't remember from where or when I started using it but it stuck

**Review by seanspeaks — APPROVED** [review]
Looks good to me; I would love for @MichaelRyanWebber To review it as well, specifically because you both have now worked on/reviewed a Microsoft Graph API-based api module, and you're both now heavily involved in API module development.

@roboli you should still merge and deploy this, but @MichaelRyanWebber Worth taking a look and noting either here or in a different issue/thread the differences between your work and Roberto's, and then let's see if we can coalesce around a standard (if there are obvious differences... otherwise, if we've already coalesced or if there are ideas to borrow between multiple, awesome, let's propagate.)

**Pull request merged**